koku/masu/database/report_manifest_db_accessor.py

```diff
@@ -80,6 +81,22 @@ def update_number_of_files_for_manifest(self, manifest):
             manifest.num_total_files = set_num_of_files
             manifest.save(update_fields=["num_total_files"])
 
+    def update_manifest_state(self, step, interval, manifest_id=None):
+        """Update the number of files for manifest."""
+        if manifest_id:
+            with transaction.atomic():  # prevent collisions
+                manifest = CostUsageReportManifest.objects.select_for_update().get(id=manifest_id)
+                if manifest:
+                    time_now = timezone.now()
+                    if not manifest.state.get(step):
+                        manifest.state[step] = {}
+                    manifest.state[step][interval] = time_now.isoformat()
+                    if interval == "end" or interval == "failed":
+                        manifest.state[step]["time_taken_seconds"] = (
+                            time_now - datetime.fromisoformat(manifest.state[step]["start"])
```

Comment:
Use the enum values here to avoid errors.
```suggestion
                    if interval == ManifestState.END or interval == ManifestState.FAILED:
                        manifest.state[step]["time_taken_seconds"] = (
                            time_now - datetime.fromisoformat(manifest.state[step][ManifestState.START])
```